qiskit/transpiler/passmanager.py

```diff
@@ -143,19 +147,56 @@ def _normalize_passes(passes):
                 raise TranspilerError('%s is not a pass instance' % pass_.__class__)
         return passes
 
-    def run(self, circuit):
-        """Run all the passes on a QuantumCircuit
+    def run(self, circuit_or_circuits):
+        """Run all the passes on circuit or circuits
 
         Args:
-            circuit (QuantumCircuit): circuit to transform via all the registered passes
+            circuit_or_circuits (QuantumCircuit or list[QuantumCircuit]): circuit(s) to
+            transform via all the registered passes.
 
         Returns:
-            QuantumCircuit: Transformed circuit.
+            QuantumCircuit or list[QuantumCircuit]: Transformed circuit(s).
         """
+        if isinstance(circuit_or_circuits, QuantumCircuit):
+            return self._run_single_circuit(circuit_or_circuits)
+        else:
+            return self._run_several_circuits(circuit_or_circuits)
+
+    def _create_running_passmanager(self):
         running_passmanager = RunningPassManager(self.max_iteration, self.callback)
         for pass_set in self._pass_sets:
             running_passmanager.append(pass_set['passes'], **pass_set['flow_controllers'])
+        return running_passmanager
+
+    @staticmethod
+    def _in_parallel(circuit, pm_dill=None):
+        """ Used by _run_several_circuits. """
+        running_passmanager = dill.loads(pm_dill)._create_running_passmanager()
```

Comment:
I guess I don't understand the need for serializing the passmanager here. Why can't this function just receive `pm` itself? I thought the point of the recent refactor was that `pm` is simple and not stateful, so why can't it just be passed to a function?

Reply:
It's coming the combination of the following elements:
 - [`Pool` pickles the parameters](https://thelaziestprogrammer.com/python/a-multiprocessing-pool-pickle). 
 - [Pickling callables might be tricky](https://medium.com/@emlynoregan/serialising-all-the-functions-in-python-cd880a63b591).

So.. serializing a pass manager needs a "smarter" serializer, like `dill`.